### PR TITLE
Issues can be closed via API (#3170)

### LINKED ISSUE #3170: FR: API to close already existing issue
Probably via edit, though it can have endpoint of its own.

### PR DESCRIPTION
**Pull request by bkcsoft**
Adds support to change an Issues status via the REST-API. (Fixes #3170)
Depends on gogits/go-gogs-client#41

(Edit: builds failing because of dependency)

### CODE REVIEW

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -161,6 +161,12 @@ func EditIssue(ctx *context.APIContext, form api.EditIssueOption) {
 		ctx.Error(500, "UpdateIssue", err)
 		return
 	}
+	if form.State != nil {
+		if err = issue.ChangeStatus(ctx.User, ctx.Repo.Repository, strings.Compare("closed", *form.State) == 0); err != nil {
```
Comment (Unknwon) [inline routers/api/v1/repo/issue.go]: I think it should compare with https://github.com/gogits/go-gogs-client/blob/master/issue.go#L16

Comment (bkcsoft) [inline routers/api/v1/repo/issue.go]: Doesn't seem like I can do that...  
`invalid type assertion: gogs.STATE_CLOSED.(string) (non-interface type gogs.StateType on left) (build)at line 165 col 1`  
`cannot use gogs.STATE_CLOSED (type gogs.StateType) as type string in argument to strings.Compare (build)at line 165 col 1`

Comment (Unknwon) [inline routers/api/v1/repo/issue.go]: `gogs.StateType(*form.State)`

Comment (bkcsoft) [inline routers/api/v1/repo/issue.go]: so I shouldn't use `strings.Compare` ?

Comment (Unknwon) [inline routers/api/v1/repo/issue.go]: I think `==` is good..

**Comment by bkcsoft** [comment]
@Unknwon done :)

**Comment by Unknwon** [comment]
Thanks!

**Pull request merged**